### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by mumo-esther**
In this PR I,
**Add the following validations:**
- For the User model:

  - Name must not be blank.
  - PostsCounter must be an integer greater than or equal to zero.

- For the Post model:

  - Title must not be blank.
  - Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.

- LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+RSpec.describe User, type: :model do
+  context 'validations' do
+    it 'is valid with valid attributes' do
+      expect(User.new(name: 'Peter')).to be_valid
+    end
+    it 'is not valid without a name' do
+      expect(User.new).to_not be_valid
+    end
+    it 'Is should valid the post counter' do
+      expect(User.new(name: 'Chendi', posts_counter: 0)).to be_valid
```
Comment (tresorsawasawa) [inline spec/models/user_spec.rb]: - [ ] According to the project requirements, Any custom methods in the models should have a corresponding unit test. Kindly make sure the `recent_posts` methods from the User model has a corresponding unit test. ✍🏻

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      expect(@post_one).to_not be_valid
+    end
+
+    it 'Should be invalid if the title length is more than 250 char' do
+      @post_one.Title = "'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
+       Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus
+        et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis,
+     ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec.'"
+      expect(@post_one).to_not be_valid
+    end
```
Comment (tresorsawasawa) [inline spec/models/post_spec.rb]: - [ ] Same here,  Any custom methods in the models should have a corresponding unit test. Kindly make sure the `update_post_counter` and `recent_comments` methods from the Post model have a corresponding unit test. ✍🏻

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    it 'is valid with valid attributes' do
+      expect(Comment.new(posts_id: 1, author_id: 1, Text: 'Hi Tom!')).to be_valid
+    end
+    it 'is not valid without a content' do
+      expect(Comment.new).to_not be_valid
+    end
+  end
+  context 'check increment' do
+    it 'Should increment by one' do
+      expect(Post.find(1).CommentsCounter).to be 1
```
Comment (tresorsawasawa) [inline spec/models/comment_spec.rb]: - [ ] Same here,  All custom methods in the models should have a corresponding unit test. let's make sure the `comment_counter` method from the Comment model have a corresponding unit test. ✍🏻  

- [ ] Please do the same for the `likes_counters` method from the Like model has a corresponding unit test. :100: :+1:

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @mumo-esther,
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @mumo-esther     :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**